model_analyzer/config/generate/manual_model_config_generator.py

```diff
@@ -67,43 +141,55 @@ def _generate_model_configs(self):
 
     def _generate_remote_mode_model_configs(self):
         """ Generate model configs for remote mode """
-        return [self._make_remote_model_config()]
+        return [[self._make_remote_model_config()]]
 
     def _generate_direct_modes_model_configs(self):
         """ Generate model configs for direct (non-remote) modes """
         model_configs = []
-        param_combos = self._get_param_combinations()
-        for param_combo in param_combos:
-            model_config = self._make_direct_mode_model_config(param_combo)
-            model_config.set_cpu_only(self._cpu_only)
+        for param_combo in self._non_max_batch_size_param_combos:
+            configs_with_max_batch_size = []
+            if self._max_batch_sizes:
+                for mbs in self._max_batch_sizes:
+                    param_combo['max_batch_size'] = mbs
+                    model_config = self._make_direct_mode_model_config(
+                        param_combo)
+                    model_config.set_cpu_only(self._cpu_only)
+                    configs_with_max_batch_size.append(model_config)
+            else:
+                model_config = self._make_direct_mode_model_config(param_combo)
+                model_config.set_cpu_only(self._cpu_only)
+                configs_with_max_batch_size.append(model_config)
 
-            model_configs.append(model_config)
+            model_configs.append(configs_with_max_batch_size)
 
         return model_configs
 
-    def _get_param_combinations(self):
+    def _determine_max_batch_sizes_and_param_combos(self):
         """
-        Calculate all parameter combinations to apply on top of the
-        base model config for manual search 
+        Determine self._max_batch_sizes and self._non_max_batch_size_param_combos 
         """
+        if self._remote_mode:
+            return
 
         if self._default_only:
-            param_combos = [DEFAULT_CONFIG_PARAMS]
+            self._non_max_batch_size_param_combos = [DEFAULT_CONFIG_PARAMS]
         else:
-            model_config_params = self._base_model.model_config_parameters()
+            model_config_params = copy(
```

Comment:
does this need to be a deepcopy?

Reply:
probably not, but I will do it to be safe